File: readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

int main(void)

{
    // Initializing variables [text] [Letters] [Sentences] [Words]
    string text = get_string("Text: ");
    int Letters = 0;
    int Sentences = 0;
    int Words = 1;

    // Iterating through each character in array [text]
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        // Counting uppercase and lowercase letters
        if ((int) text[i] >= 65 && (int) text[i] <= 90)
        Letters++;
        if ((int) text[i] >= 97 && (int) text[i] <= 122)
        Letters++;

        // Counting sentences
        if ((int) text[i] == 46 || (int) text[i] == 33 || (int) text[i] == 63)
        Sentences++;

        // Counting words
        if ((int) text[i] == 32)
        Words++;

    }

    // Computing values [L] and [S]
    float L = Letters * 100 / Words;
    float S = Sentences * 100 / Words;
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    // Computing CL index, with conditions for extremeties
    printf("Coleman-Liau readability index:\n");
    if (index >= 16)
    printf("Grade 16+");
    else if (index <1)
    printf("Before Grade 1");
    else
    printf("Grade %i" , index);
}

File: readability.py
from cs50 import get_string

# Requesting text to be analyzed, initializing counters
l = 0
s = 0
w = 1
text = get_string("Text: ")

# Iterating through each character

for i in range(len(text)):

    # Counting uppercase letters
    if (ord(text[i]) >= 65 and ord(text[i]) <= 90):
        l += 1

    # Counting lowercase letters
    if (ord(text[i]) >= 97 and ord(text[i]) <= 122):
        l += 1

    # Counting sentences
    if (ord(text[i]) == 46 or ord(text[i]) == 33 or ord(text[i]) == 63):
        s += 1

    # Counting words
    if (ord(text[i]) == 32):
        w += 1

# Computing values L and S for CL index
L = (float(l) * 100 / w)
S = (float(s) * 100 / w)
index = round(0.0588 * L - 0.296 * S - 15.8)

# Expressing CL index
if (index >= 16):
    print("Grade 16+")
elif (index < 1):
    print("Before Grade 1")
else:
    print(f"Grade {index}")